rcp/org.testeditor.rcp4.tests/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialogTest.xtend

```diff
@@ -0,0 +1,164 @@
+		// given
+		initPrefsMockWithProxy()
+		when(prefs.get("http.proxyHost", "")).thenReturn("proxy")
+		System.setProperty("http.proxyHost", "myProxy")
+		initContextWithProxyMock()
+		var NetworkConnectionSettingDialog dialog = new NetworkConnectionSettingDialog(parentShell, prefs, context)
+		// when
+		assertTrue(dialog.isInternetAvailable(true))
+		// then
+		assertEquals("proxy", System.getProperty("http.proxyHost"))
+	}
+
+	@Test def void testExtractProxySettingsFromMavenFile() throws Exception {
+		// Given
+		initContextWithProxyMock()
+		System.setProperty("http.proxyHost", "")
+		val NetworkConnectionSettingDialog dialog = new NetworkConnectionSettingDialog(parentShell, prefs, context)
+		val File mavenSettingsFile = tempFolder.newFile("settings.xml")
+		val content = '''
+		<?xml version="1.0" encoding="UTF-8"?>
+		<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+			xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+			xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
+			https://maven.apache.org/xsd/settings-1.0.0.xsd">
+			<proxies>
+				<proxy>
+					<id>myproxy</id>
+					<active>true</active>
+					<protocol>http</protocol>
+					<host>proxy.somewhere.com</host>
+					<port>8080</port>
+					<username>proxyuser</username>
+					<password>somepassword</password>
+					<nonProxyHosts>*.google.com|ibiblio.org</nonProxyHosts>
+				</proxy>
+			</proxies>
+		</settings>'''
+		Files.write(content, mavenSettingsFile, UTF_8);
+
+		// when
+		dialog.extractProxyFromMavenSettings(mavenSettingsFile)
+		dialog.isInternetAvailable(true)
+
+		// then
+		assertEquals("proxy.somewhere.com", System.getProperty("http.proxyHost"))
+	}
+
+	@Test def void testNoProxySettingsiNMavenFile() throws Exception {
+		// Given
+		System.setProperty("http.proxyHost", "")
+		initContextWithProxyMock()
+		val NetworkConnectionSettingDialog dialog = new NetworkConnectionSettingDialog(parentShell, prefs, context)
+		val File mavenSettingsFile = tempFolder.newFile("settings.xml")
+		val content = '''
+		<?xml version="1.0" encoding="UTF-8"?>
+		<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+			xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+			xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
+			https://maven.apache.org/xsd/settings-1.0.0.xsd">
+		</settings>'''
+		Files.write(content, mavenSettingsFile, UTF_8);
+
+		// when
+		dialog.extractProxyFromMavenSettings(mavenSettingsFile)
+		dialog.isInternetAvailable(true)
+
+		// then
+		assertEquals("", System.getProperty("http.proxyHost"))
+	}
+
+	@Test def void testnOProxySettingsiNMavenFile() throws Exception {
+		// Given
+		val NetworkConnectionSettingDialog dialog = new NetworkConnectionSettingDialog(parentShell, prefs, context)
+		val mavenSettingsFile = mock(File);
+		when(mavenSettingsFile.exists).thenReturn(false);
+
+		// when
+		try {
+			dialog.extractProxyFromMavenSettings(mavenSettingsFile)
+		} catch (Exception e) {
+			fail(e.message)
```

Comment:
I see it is to short with the error message of the exception. The purpose was to communicate in the test code, that the tested method should not throw an exception on a missing file.